### PR TITLE
Redis json

### PR DESCRIPTION
**Pull request by brenopessoa**
https://github.com/eclipse/jnosql-diana-driver/issues/30

### CODE REVIEW

File: redis-driver/pom.xml [hunk 1]
```diff
@@ -66,13 +66,22 @@
...
-        <dependency>
-            <groupId>junit</groupId>
-            <artifactId>junit</artifactId>
-            <version>4.12</version>
-            <scope>test</scope>
-        </dependency>
-        <dependency>
+    	 
+    	 <dependency>
+             <groupId>javax.json</groupId>
```
Comment (otaviojava) [inline redis-driver/pom.xml]: Once these dependencies are in the parent we don't need to replace this:
https://github.com/eclipse/jnosql-diana-driver/blob/master/pom.xml#L91

File: redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisCollection.java [hunk 2]
```diff
@@ -32,13 +34,13 @@
 
     protected Jedis jedis;
 
-    protected Gson gson;
+    protected Jsonb JsonB;
```
Comment (otaviojava) [inline redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisCollection.java]: please, rename this variable to `jsonB`

File: redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisKeyValueEntityManager.java [hunk 3]
```diff
@@ -16,31 +16,34 @@
...
+import redis.clients.jedis.Jedis;
 
 /**
  * The redis implementation to {@link BucketManager}
  */
 public class RedisKeyValueEntityManager implements BucketManager {
 
     private final String nameSpace;
-    private final JSONValueProvider provider;
+    private final Jsonb provider;
```
Comment (otaviojava) [inline redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisKeyValueEntityManager.java]: Rename to `jsonB`

File: redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisKeyValueEntityManagerFactory.java [hunk 4]
```diff
@@ -15,23 +15,25 @@
...
+
+import redis.clients.jedis.JedisPool;
+
 /**
  * The redis implementation to {@link BucketManagerFactory} where returns {@link RedisKeyValueEntityManager}
  */
 public class RedisKeyValueEntityManagerFactory implements BucketManagerFactory<RedisKeyValueEntityManager> {
 
-    private static final JSONValueProvider PROVDER = JSONValueProviderService.getProvider();
+    private static final Jsonb PROVDER = JsonbBuilder.create();
```
Comment (otaviojava) [inline redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisKeyValueEntityManagerFactory.java]: Please rename that to `JSON`

File: redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisMap.java [hunk 5]
```diff
@@ -38,15 +40,15 @@
 
     private final Jedis jedis;
 
-    private final Gson gson;
+    protected Jsonb JsonB;
```
Comment (otaviojava) [inline redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisMap.java]: Please, rename to `jsonB`

**Comment by brenopessoa** [comment]
Fix the PR with the suggestions

File: redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisCollection.java [hunk 6]
```diff
@@ -32,13 +34,13 @@
 
     protected Jedis jedis;
 
-    protected Gson gson;
+    protected Jsonb jsonB;
```
Comment (otaviojava) [inline redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisCollection.java]: @brenopessoa could you check if this attribute can be static, please?

File: redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisKeyValueEntityManager.java [hunk 7]
```diff
@@ -16,33 +16,36 @@
...
+import redis.clients.jedis.Jedis;
 
 /**
  * The redis implementation to {@link BucketManager}
  */
 public class RedisKeyValueEntityManager implements BucketManager {
 
     private final String nameSpace;
-    private final JSONValueProvider provider;
+    private final Jsonb jsonB;
```
Comment (otaviojava) [inline redis-driver/src/main/java/org/jnosql/diana/redis/key/RedisKeyValueEntityManager.java]: Jsonb is Thread safe, maybe convert this attribute to static

**Comment by otaviojava** [comment]
@brenopessoa Thanks for the PR.

**Pull request merged**